pattoo/api/portal/routes/agents.py

```diff
@@ -0,0 +1,66 @@
+"""Routes for agents."""
+
+# Flask imports
+from flask import Blueprint
+from flask import request, session
+
+# Pattoo imports
+from pattoo.db import db
+from pattoo.db.models import Agent as AgentModel
+
+# Define the AGENTS global variable
+AGENTS = Blueprint('AGENTS', __name__)
+
+@AGENTS.route('/api/agent', methods=['GET'])
+def agents():
+    """Agents route.
+    
+    Args:
+        None
+
+    Returns:
+        response (dict): Response Message
+
+    """
+    # Check if a user was stored in session first
+    if session.get('idx_user', None) is None:
+        response = {'data': {'message': 'Login first'}}
+        # Block access if no user was found in session
+        return response
+
+    enable = request.args.get('enable', type=int)
+    idx_agent = request.args.get('agent', type=int)
+
+    response = {'data':{'message': 'Query did not run'}}
+
+    if enable is not None and idx_agent is not None:
+        # Toggle the enable status from one received from request
+        change_enable = 1 if (enable == 0) else 0
+        response = {'data': {'idx_agent': idx_agent, 'enable': enable, 'message': 'Not changed'}}
+        with db.db_modify(20188, die=True) as db_session:
+            db_session.query(AgentModel).filter(
+                AgentModel.idx_agent == idx_agent
+            ).update({'enabled': change_enable})
+            # Return successful message
+            response = {'data': {'idx_agent': idx_agent, 'enable': change_enable, 'message': 'Changed'}}
```

Comment:
Is there any way to create unique messages that include a code number? This will make troubleshooting easier. It makes the creation of user guides easier in future. How would you enforce that the codes are unique?